addons/binding/org.openhab.binding.russound/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,324 @@
+
+    <bridge-type id="rio">
+        <label>Russound RIO Device</label>
+        <description>Ethernet access point to Russound RIO control system (usually the main controller)</description>
+        
+        <channels>
+            <channel id="version" typeId="readonlystring"><label>Firmware Version</label><description>Firmware Version</description></channel>
+            <channel id="lang" typeId="lang"/>
+            <channel id="allon" typeId="button"><label>All Zones</label><description>Toggles All Zones</description></channel>
+        </channels>
+
+        <config-description>
+            <parameter name="ipAddress" type="text" required="true">
+                <context>network-address</context>
+                <label>IP or Host Name</label>
+                <description>The IP or host name of the Russound RIO access point</description>
+            </parameter>
+            <parameter name="ping" type="integer" required="false">
+                <label>Ping Interval</label>
+                <description>The ping interval in seconds to keep the connection alive</description>
+                <default>30</default>
+            </parameter>
+            <parameter name="retryPolling" type="integer" required="false">
+                <label>Retry Polling</label>
+                <description>The polling, in seconds, to retry a connection attempt</description>
+                <default>10</default>
+            </parameter>
+        </config-description>
+        
+    </bridge-type>
+
+    <bridge-type id="controller">
+        <supported-bridge-type-refs>
+            <bridge-type-ref id="rio"/>
+        </supported-bridge-type-refs>
+
+        <label>Russound Controller</label>
+        <description>Controller of Zones, Sources, etc</description>
+
+        <channels>
+            <channel id="type" typeId="readonlystring"><label>Model Type</label><description>Model Type</description></channel>
+            <channel id="ipaddress" typeId="readonlystring"><label>IP Address</label><description>IP Address</description></channel>
+            <channel id="macaddress" typeId="readonlystring"><label>MAC Address</label><description>MAC Address</description></channel>
+        </channels>
+        
+        <config-description>
+            <parameter name="controller" type="integer" min="1" max="6" required="true">
+                <label>Controller ID</label>
+                <description>The controller identifier</description>
+            </parameter>
+        </config-description>
+    </bridge-type>
+
+    <bridge-type id="zone">
+        <supported-bridge-type-refs>
+            <bridge-type-ref id="controller"/>
+        </supported-bridge-type-refs>
+        
+        <label>Russound Zone</label>
+        <description>Zone within a Controller</description>
+        
+        <channels>

```

Comment:
make sure that only a few are regular channels and the majority are advanced channels.